compiler-plugin/src/main/java/io/ballerina/stdlib/data/jsondata/compiler/JsondataCodeAnalyzer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.data.jsondata.compiler;
+
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.projects.plugins.CodeAnalysisContext;
+import io.ballerina.projects.plugins.CodeAnalyzer;
+
+import java.util.List;
+
+/**
+ * Jsondata Code Analyzer.
+ */
+public class JsondataCodeAnalyzer extends CodeAnalyzer {
+
+        @Override
+        public void init(CodeAnalysisContext codeAnalysisContext) {
+            codeAnalysisContext.addSyntaxNodeAnalysisTask(new JsondataRecordFieldValidator(),
+                    List.of(SyntaxKind.MODULE_PART));
+        }
```

Comment:
It is better this can be indent to left side ( a tab).
Similar apply for JsondataCompilerPlugin